atomic_reactor/plugins/pre_resolve_remote_source.py

```diff
@@ -129,6 +132,7 @@ def set_worker_params(self, source_request, remote_source_url, remote_source_con
         override_build_kwarg(self.workflow, 'remote_source_url', remote_source_url)
         override_build_kwarg(self.workflow, 'remote_source_build_args', build_args)
         override_build_kwarg(self.workflow, 'remote_source_configs', remote_source_conf_url)
+        override_build_kwarg(self.workflow, 'remote_source_request_id', remote_source_request_id)
```

Comment:
Could we do it as we do above with the configuration files and pass the content-manifest url? Then the download plugin will not need to deal with the cacito API client again.

The cachito response will also include a `content_manifest` field analogous to the configuration_files one: https://github.com/release-engineering/cachito/pull/187/files#diff-7cbde6013c2753c0bb969ab72bd9d02eR499. You could either start using it now or just build the URL here and we can change it after the cachito PR is merged.

Reply:
@athos-ribeiro; I must be missing something. The way I'm reading this, I'm supposed to get the ICM URL from the original request, and then essentially do the work which currently happens in`CachitoAPI::get_icm()` in the 'add_image_content_manifest' plugin instead... otherwise, what am I supposed to do with just the request URL?

*Somewhere* we have to fetch the actual ICM using the 'content_manifest' URL. We can't fetch the *content* and then pass it around (too much data), but it doesn't make sense to me to do the `request` work in the 'add_image_content_manifest' plugin vs. doing it in 'cachito.py' - seems like the actual Cachito API access should happen in the latter...

Reply:
I am suggesting to do the same thing as the we do for the `remote_source_conf_url`: What we will be passing around is a URL. The `add_image_content_manifest` plugin will not care if it comes from cachito or not. It will just download the contents in that URL and verify it is a valid ICM.

> seems like the actual Cachito API access should happen in the latter...

It is happening there, since the URL will come from the cachito request response.

How does that sound?